### PR TITLE
ExpressRoute private peering with VNet

### PR DESCRIPTION
**Pull request by fabferri**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [X ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog
First ARM template submission to create ExpressRoute circuit, VNet with ExpressRoute gateway a a connection to link the ExpressRoute Gateway to the ExpressRoute circuit.
*
*
*

### CODE REVIEW

**Comment by fabferri** [comment]
AzQuickStarts-USGov fails becasue the my paramenter file set as location  'eastus', for public cloud.
Anyone knows how to fix the check? In ARM parametners file I can reference only a single specific region in public cloud or usgov cloud.

New-AzResourceGroupDeployment : 8:15:35 PM - Error: Code=LocationNotAvailableForResourceType; Message=The provided location 'eastus' is not available for resource type 'Microsoft.Network/expressRouteCircuits'. List of available regions for the resource type is 'usgoviowa,usgovvirginia,usdodcentral,usdodeast,usgovtexas,usgovarizona'.
At D:\a\1\test\Deploy-AzTemplate.ps1:211 char:9

**Comment by fabferri** [comment]
my check fails for setting resourceGroup().location as default value value of  location:
  azuredeploy.json must use the location parameter, not resourceGroup().location (except when used as a default value in the main template)
why the templates fails?

**Comment by fabferri** [comment]
#sign-off

**Comment by fabferri** [comment]
Hello @bmoore-msft this one is ready to go for merge.
Thank you.

File: 101-expressroute-private-peering-vnet/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+        "description": "name of the subnet"
+      }
+    },
+    "vnetAddressSpace": {
+      "type": "string",
+      "metadata": {
+        "description": "address space assigned to the VNet"
+      }
+    },
+    "subnet1Prefix": {
```
Comment (bmoore-msft) [inline 101-expressroute-private-peering-vnet/azuredeploy.json:98]: add a defaultValue

File: 101-expressroute-private-peering-vnet/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+      "type": "string",
+      "metadata": {
+        "description": "network prefix assigned to the subnet"
+      }
+    },
+    "gatewaySubnetPrefix": {
```
Comment (bmoore-msft) [inline 101-expressroute-private-peering-vnet/azuredeploy.json:104]: add a defaultValue

File: 101-expressroute-private-peering-vnet/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+      "type": "string",
+      "metadata": {
+        "description": "network prefixes assigned to the gateway subnet. it has to be a network prefix with mask /27 or larger"
+      }
+    },
+    "gatewayName": {
```
Comment (bmoore-msft) [inline 101-expressroute-private-peering-vnet/azuredeploy.json:110]: add a defaultValue

File: 101-expressroute-private-peering-vnet/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+      "name": "[variables('erCircuitName')]",
+      "location": "[variables('erlocation')]",
+      "comments": "Create the ER circuit",
+      "sku": {
+        "name": "[variables('erSKU_Name')]",
+        "tier": "[variables('erSKU_Tier')]",
+        "family": "[variables('erSKU_Family')]"
+      },
+      "properties": {
+        "serviceProviderProperties": {
```
Comment (bmoore-msft) [inline 101-expressroute-private-peering-vnet/azuredeploy.json:170]: these should be parameters - not hard coded - can use allowedValues to restrict them if needed

File: 101-expressroute-private-peering-vnet/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+    "gatewaySubnetRef": "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('vnetName'), 'GatewaySubnet')]",
+    "nsg": "nsg"
+  },
+  "resources": [
+    {
+      "apiVersion": "2019-12-01",
+      "type": "Microsoft.Network/expressRouteCircuits",
+      "name": "[variables('erCircuitName')]",
+      "location": "[variables('erlocation')]",
+      "comments": "Create the ER circuit",
```
Comment (bmoore-msft) [inline 101-expressroute-private-peering-vnet/azuredeploy.json:163]: move comment properties to the top of the resource definition (see contribution guide for sort order)

File: 101-expressroute-private-peering-vnet/azuredeploy.parameters.json [hunk 6]
```diff
@@ -0,0 +1,39 @@
+{
```
Comment (bmoore-msft) [inline 101-expressroute-private-peering-vnet/azuredeploy.parameters.json:1]: remove params from here once you have defaultValues

**Comment by fabferri** [comment]
Hello @bmoore-msft  thanks for the notes.
I have changed the template based on your comments. 
Would you please make a sanity check?
Thank you

**Comment by fabferri** [comment]
Hello @bmoore-msft, 
I have tried to restore the file 101-expressroute-circuit-create/metadata.json
that  I deleted unluckily from another project but it fails. 
commit e160d702c0eb20fee530802c4e41ea0a601aa530
can you please advice how to fix it?

**Pull request merged**